src/to-have-prop.js

```diff
@@ -18,7 +18,8 @@ export function toHaveProp(element, name, expectedValue) {
   const prop = element.props[name];
 
   const isDefined = expectedValue !== undefined;
-  const isAllowed = VALID_ELEMENTS.includes(element.type);
+  const elementType = typeof element.type == 'string' ? element.type : element.type?.displayName;
+  const isAllowed = VALID_ELEMENTS.includes(elementType);
```

Comment:
I'm wondering whether we should have a list of allowed components to check their props. IMO we should be able to check props of any host component. @thymikee, @brunohkbx, wdyt?

Reply:
Same. A list of valid elements is a remnant of the previous lib implementation

Reply:
@mdjastrzebski yeah I completely agree.